processor/src/test/java/org/mapstruct/ap/test/value/string2enum/StringToEnumMappingTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.value.string2enum;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.test.value.OrderType;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.compilation.annotation.CompilationResult;
+import org.mapstruct.ap.testutil.compilation.annotation.Diagnostic;
+import org.mapstruct.ap.testutil.compilation.annotation.ExpectedCompilationOutcome;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@IssueKey( "1557" )
+@WithClasses({ OrderType.class, OrderMapper.class })
+@RunWith(AnnotationProcessorTestRunner.class)
+public class StringToEnumMappingTest {
+
+    @Test
+    public void testTheNormalStuff() {
```

Comment:
I think that for this we can also test that the exception is thrown for an unknown `String`

Reply:
Hmm.. this is actually an interesting case. In the past we added this one for the possibility that a library containing the enum is updated runtime (and a new enum value has been added), but not yet compile time.

But what should we do for when the input is a `String`? I guess an `IllegalArgument` is justified. Or perhaps we should demand an `ANY_UNMAPPED` and even give an error when it's absent.. ?

Reply:
I would say it is up to the user to define what they want. Throwing the exception is completely OK.

Reply:
> But what should we do for when the input is a `String`? I guess an `IllegalArgument` is justified. Or perhaps we should demand an `ANY_UNMAPPED` and even give an error when it's absent.. ?

thinking more about this.. This is also not the way forward. `ANY_UNMAPPED` stops after the defined mappings. Making it mandatory means the user always have to specify the mappings. 

Back to `ANY_REMAINING` and `ANY_UNMAPPED` for the string-to-enum case. And making one of these mandatory in this scenario.

Reply:
done.

Reply:
> I would say it is up to the user to define what they want. Throwing the exception is completely OK.

Generating an `IllegalArgumentException` for this case might surprise the user when running the code. A String not matching the enum constant will be pretty common. 

I prefer to make the user aware that he should make a choice here at the expense of a little bit more verbosity.

Reply:
I would say that having a compile error is too strict. Perhaps only a warning so they are aware about it. Perhaps they want to do it on purpose like that.